python/kserve/kserve/storage/storage.py

```diff
@@ -139,6 +139,21 @@ def _update_with_storage_spec():
                     f.write(value)
                     f.flush()
 
+        if storage_secret_json.get("type", "") == "gs":
+            temp_dir = tempfile.mkdtemp()
```

Comment:
Can you explain how this works, why do we write to the temp directory?

Reply:
[Google ADC](https://cloud.google.com/docs/authentication/application-default-credentials) uses `GOOGLE_APPLICATION_CREDENTIALS` env var to obtain location of the credential JSON file

Reply:
As @israel-hdez mentioned as well, from the same doc, SA is supported, https://cloud.google.com/docs/authentication/application-default-credentials#attached-sa

imho, that is the best option.

Reply:
Guess we need to add docs for this.

Reply:
Thanks @spolti and @israel-hdez for the feedback.

@spolti [docs](https://github.com/kserve/website/pull/351)

Reply:
Thanks @tjandy98 :)